playbooks/tasks/lastpass-load-webportal-config.yml

```diff
@@ -1,12 +1,32 @@
 ---
 
-# Load Portal server and common configs from LastPass
+# Load portal configs from LastPass
 
-- name: Include loading LastPass portal server config
-  include_tasks: tasks/lastpass-load-webportal-server-config.yml
+# Portal common and server configs are loaded and set always.
+# If a portal is part of a cluster:
+# - cluster config is loaded and set
+# - cluster config data are merged into and can override common config data
+# - common config then contans valid common and cluster data
```

Comment:
This is a little confusing. If I'm reading this correctly, then the configuration is split into two parts - `common` config (shared by all servers) and `server` config (per server).

My question is: Shouldn't this setup be used only when the portal is using a cluster of servers? Otherwise the `common` and `server` configs will be identical, right?

In that case isn't the `cluster` config the same as the `common` config? Or do we have all three?

Example:
`common` says "A: 1"
`cluster` says "A: 2" -> overrides `common` and sets A to 2
`server` says "A: 3" -> overrides all and sets A to 3

Reply:
There will be
* a generic portal setup Ansible role (WIP)
* a Skynet Labs specific portal setup Ansible playbook using this role

Generic role requires (implementation is upto operator, they do not have to use LastPass)
* generic load `common` config var and `server` config var
* generic save `server` config var

Skynet Labs specific playbook loads and saves to LastPass.

SkynetLabs has 3 types configs in LastPass:
* common - 1x
* cluster - currently 2x: prod, dev-a
* server - for each server
Each LastPass config has different data, they do not overlap.

In Skynet Labs Ansible playbook we have 3 config vars:
* common
* cluster
* server

Skynet Labs Ansible playbook does for each host (portal):
* `common` var loads `common` data from LastPass
* `cluster` var loads `cluster` data from LastPass
* `common` var is set to have `common` data overriden with `cluster` data from `cluster` var
* `cluster` var is set to null, because we have now merged data in `common` var for a specific server and Ansible role doesn't use `cluster` var
* `server` var loads `server` data from LastPass (and doesn't override anything)

We can change common settings by just editing `common` or `cluster` configs in LastPass and (WIP) rerunning portal setup script and we are sure they will be set on all servers, no server can override them by its own `server` data from LastPass.

Operator with just 1 server needs 1x `common` and 1x `server` configs data in their backend, they will not overlap.

Reply:
OK. I was missing the part where the data don't overlap.